packages/mobile/src/invite/saga.ts

```diff
@@ -286,25 +296,32 @@ export function* redeemInviteSaga({ inviteCode }: RedeemInviteAction) {
 
   const {
     result,
+    cancel,
     timeout,
   }: {
-    result: { success: true; newAccount: string } | { success: false }
-    timeout: true
+    result: { success: true; newAccount: string } | { success: false } | undefined
+    cancel: TakeEffect | undefined
+    timeout: true | undefined
   } = yield race({
     result: call(doRedeemInvite, inviteCode),
+    cancel: take(Actions.CANCEL_REDEEM_INVITE),
     timeout: delay(REDEEM_INVITE_TIMEOUT),
   })
 
-  if (result.success === true) {
+  if (result?.success === true) {
     Logger.debug(TAG, 'Redeem Invite completed successfully')
     yield put(redeemInviteSuccess())
     navigate(Screens.VerificationEducationScreen)
     // Note: We are ok with this succeeding or failing silently in the background,
     // user will have another chance to register DEK when sending their first tx
     yield spawn(registerAccountDek, result.newAccount)
-  } else if (result.success === false) {
+  } else if (result?.success === false) {
     Logger.debug(TAG, 'Redeem Invite failed')
     yield put(redeemInviteFailure())
+  } else if (cancel) {
+    ValoraAnalytics.track(OnboardingEvents.invite_redeem_cancel)
+    yield put(redeemInviteFailure())
```

Comment:
Following what I said above, I think we shouldn't dispatch a failure here, it's not really a failure, it's just the user who decided to cancel. The reducer should update the state for the `CANCEL_CREATE_OR_RESTORE_ACCOUNT` action.